### PR TITLE
Test against Milvus 2.4

### PR DESCRIPTION
**Pull request by roji**

### CODE REVIEW

**Comment by roji** [comment]
Note: after this is merged, main will represent the 2.4 version of the SDK. I'll create a release/2.3 branch so that we can continue improving the 2.3 SDK as necessary, e.g. to release 2.3.1.

File: Milvus.Client.Tests/CollectionTests.cs [hunk 1]
```diff
@@ -287,8 +287,13 @@ public async Task Compact()
             });
 
         long compactionId = await collection.CompactAsync();
-        Assert.NotEqual(0, compactionId);
+        if ((await Client.GetVersionAsync()).StartsWith("v2.4.", StringComparison.Ordinal))
+        {
+            // Milvus 2.4 returns -1 here as the compaction ID
```
Comment (roji) [inline Milvus.Client.Tests/CollectionTests.cs:292]: @yhmo @weianweigan FYI, this is the only case where a test failed when transitioning from 2.3.10 to 2.4.0-rc.1 - not sure if this is intentional...

**Review by weianweigan — APPROVED** [review]

File: Milvus.Client.Tests/CollectionTests.cs (hunk 1 above)
Comment (weianweigan) [reply Milvus.Client.Tests/CollectionTests.cs:292]: Let me try to contact @yhmo .

**Comment by roji** [comment]
Note that I've pushed branch release/2.3 for any work that would happen for 2.3.

**Comment by yhmo** [comment]
A bug of compaction is fixed in milvus v2.4.0: https://github.com/milvus-io/milvus/issues/29943
According to the issue description. The old version behavior is: **CompactionPlanCount is never set, but CompactionID is set a value regardless of whether a plan is generated or not. This makes it difficult for users to determine whether a compaction plan has been generated.**

In v2.4.0, the ManualCompaction() interface has been changed: **CompactionPlanCount should be set to the count of the generated compaction plan and CompactionID should be set to a negative value if there's no plan generated.**

**Pull request merged**